### PR TITLE
Add strict-origin and strict-origin-when-cross-origin policy states (fixes #50)

### LINKED ISSUE #50: Add the non-downgrading versions of origin and origin-when-cross-origin
As discussed [on the list](https://lists.w3.org/Archives/Public/public-webappsec/2016Mar/0085.html), in the [last teleconf](https://www.w3.org/2016/04/20-webappsec-minutes.html#item09) and in the Face-to-face, I will follow up #19 with spec changes for the other two proposed policy states.

### PR DESCRIPTION
**Pull request by fmarier**
Since @mozfreddyb  is out of the office until September, I'll take over his patch and so this pull request supersedes #52.

The three commits I added should address the issues that @estark37 found.

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -332,6 +332,42 @@ <h3 dfn export id="referrer-policy-origin" oldids="referrer-policy-state-origin"
     priori</em> authenticated URLs</a>.
   </div>
 
+  <h3 dfn export id="referrer-policy-safe-origin" oldids="referrer-policy-state-safe-origin">"<code>safe-origin</code>"</h3>
+
+  The <a>"<code>safe-origin</code>"</a> policy sends the
+  <a lt="ASCII serialization of an origin">ASCII serialization</a> of the
+  <a>origin</a> of the <a>request client</a> when making requests from a
+  <a>TLS-protected</a> <a>environment settings object</a> to a
+  <a><em>a priori</em> authenticated URL</a>, and requests from
```
Comment (estark37) [inline index.src.html]: Would you be okay with changing "..., and requests from" to "..., and when making requests from"? To me that makes it a little easier to parse.

**Comment by estark37** [comment]
@fmarier thanks for updating this and for the examples in https://github.com/w3c/webappsec-referrer-policy/pull/52#issuecomment-229515024.

I'd be fine with `strict-` and I like that better than `safe-`. Mostly for completeness, I'll suggest that we could be more explicit and do something like `origin-when-not-downgrading` and `origin-when-cross-origin-not-downgraded` or something like that... but that's super long and also grammatically awkward so `strict-` is probably better.

**Comment by fmarier** [comment]
> Would you be okay with changing "..., and requests from" to "..., and when making requests from"? To me that makes it a little easier to parse.

I've turned that bit into a bullet list. Do you think that's easier to read?

The actual wording is still pretty dense. It would be great if we could say "HTTPS page loading HTTP resource" in less words than what we currently have.

**Comment by estark37** [comment]
Thanks @fmarier, merged! I think the bulleted list makes it a lot easier to read.

**Comment by estark37** [comment]
And thanks @mozfreddyb too.

**Pull request merged**